pkgs/by-name/se/segger-systemview/package.nix

```diff
@@ -0,0 +1,188 @@
+{ lib
+, stdenv
+, fetchurl
+, autoPatchelfHook
+, segger-jlink
+, config
+, acceptLicense ? config.segger-systemview.acceptLicense or false
+, fontconfig
+, xorg
+, makeDesktopItem
+, copyDesktopItems
+}:
+
+let
+  supported = {
+    x86_64-linux = {
+      name = "x86_64";
+      sha256 = "d3ec8d54ae0605a40d12d6eeea80967362ed3ed921395f154efee8d65d0c25ee";
+    };
+    i686-linux = {
+      name = "i386";
+      sha256 = "2292a6ed6c9b277e514ddbafb396e0e5a8ef8ca57ebd3236cf5ae29a52d2b5bc";
+    };
+  };
+
+  platform = supported.${stdenv.targetPlatform.system} or (throw "unsupported platform ${stdenv.targetPlatform.system}");
+
+  version = "352a";
+
+  url = "https://www.segger.com/downloads/systemview/SystemView_Linux_V${version}_${platform.name}.tgz";
+
+  src =
+    assert !acceptLicense -> throw ''
+      Use of the "SEGGER SystemView real-time software analysis too" requires the
+      acceptance of the following licenses:
+
+        - SEGGER SystemView Installation and Activation [1]
+        - SEGGER Software Licensing [2]
+
+      You can express acceptance by setting acceptLicense to true in your
+      configuration. Note that this is not a free license so it requires allowing
+      unfree licenses as well.
+
+      configuration.nix:
+        nixpkgs.config.allowUnfree = true;
+        nixpkgs.config.segger-systemview.acceptLicense = true;
+
+      config.nix:
+        allowUnfree = true;
+        segger-systemview.acceptLicense = true;
+
+      [1]: https://www.segger.com/products/development-tools/systemview/license/systemview-installation/
+      [2]: https://www.segger.com/purchase/licensing/
+    '';
+      fetchurl {
+        inherit url;
+        inherit (platform) sha256;
+      };
+
+  qt4-bundled = stdenv.mkDerivation {
```

Comment:
We don't accept more qt4 package and have been trying to remove or update them.

Reply:
The package has a hard dependency on QT4. Without QT4, it does not run. Please review the discussion on QT4 at https://github.com/NixOS/nixpkgs/pull/255185 , SEGGER is unwilling to upgrade the QT version.

Reply:
So to introduce a proprietary software with outdated dependencies... why? It's not like it was ever there and we need to remove it.

Reply:
I understand your concerns, but as @h7x4 pointed out, this package required the user to explicitly understand and allow (1) the license and (2) the outdated libs. Other packages and systems are unaffected by this package, i.e. if you do not use it you are unaffected.

SystemView is a widely used industry-standard embedded debugging suite, see https://www.segger.com/products/development-tools/systemview/ . It would be nice for NixOS to evolve from a project for ambitions hobbyists into a system which supports real-world software. 

If I could, I would just provide this package out-of tree, similar to how e.g. Arch (https://aur.archlinux.org/packages/jlink-systemview) does it, but this does not work for nixpkgs. I do not have the bandwidth to maintain an eternal fork for a simple package.

Reply:
Can we dedupe this with https://github.com/NixOS/nixpkgs/blob/master/pkgs/by-name/se/segger-jlink/qt4-bundled.nix ?

Reply:
> If I could, I would just provide this package out-of tree, similar to how e.g. Arch ([aur.archlinux.org/packages/jlink-systemview](https://aur.archlinux.org/packages/jlink-systemview)) does it, but this does not work for nixpkgs. I do not have the bandwidth to maintain an eternal fork for a simple package.

There is the https://github.com/nix-community/NUR.